tests/run-duration.feature

```diff
@@ -0,0 +1,19 @@
+# This file is part of REANA.
+# Copyright (C) 2023 CERN.
+#
+# REANA is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+Feature: Run duration
+
+    As a researcher,
+    I want to assert that my workflows run in a reasonable amount of time,
+    If they take too long, it's a sign that something in the workflow is not
+    working as expected.
+
+    Scenario: The workflow terminates in a reasonable amount of time
+        When the workflow finishes
+        Then the status should be "finished"
+        And the workflow run duration should be less than 10 minutes
```

Comment:
Perhaps split into three scenarios? First for the whole workflow time estimate, second for the gendata step, third for the fitdata step.